warehouse/query-core/src/main/java/datawave/query/iterator/logic/TermFrequencyExcerptIterator.java

```diff
@@ -315,28 +386,244 @@ public void next() throws IOException {
+            }
+            if (startNull) {
+                termsToOutput[k] = null;
+            } else {
+                if (termsToOutput[k] == null) {
+                    start++;
+                }
+            }
+        }
+    }
+
+    /**
+     * Looks for hit phrases (not separate hit terms) and puts the whole phrase in brackets
+     *
+     * @param terms
+     *            the terms to create a phrase from
+     */
+    private void checkForHitPhrase(WordsAndScores[] terms) {
+        ArrayList<String> hitPhrases = new ArrayList<>();
+        // checks for phrases (anything in the hit list with a space in it) and adds them to a new arrayList
+        for (String s : hitTermsList) {
+            if (s.contains(" ")) {
+                hitPhrases.add(s);
+            }
+        }
+        // if we don't find any, return unchanged
+        if (hitPhrases.isEmpty()) {
+            return;
+        }
+        // for each hit phrase found...
+        for (String hitPhrase : hitPhrases) {
+            // split the phrase on the spaces into the separate terms
+            String[] individualHitTerms = hitPhrase.split(" ");
+            // if the phrase is almost the same size as the whole excerpt, skip this iteration
+            if ((terms.length - 2) < individualHitTerms.length) {
+                continue;
+            }
+            // iterate across the WordsAndScores until the end of the hit phrase reaches the last offset
+            int iterations = terms.length - individualHitTerms.length + 1;
+            for (int j = 0; j < iterations; j++) {
+                // if we find the hit phrase...
+                if (isPhraseFound(individualHitTerms, terms, j)) {
+                    // set which position in the phrase each offset is
+                    int overridePosition;
+                    for (int k = 0; k < individualHitTerms.length; k++) {
+                        // beginning of phrase
+                        if (k == 0) {
+                            overridePosition = 1;
+                        } else if (k == individualHitTerms.length - 1) { // end of phrase
+                            overridePosition = 3;
+                        } else { // middle of phrase
+                            overridePosition = 2;
+                        }
+                        // set the override values for the current positions WordsAndScores to the index of the hit term in this position plus the override
+                        terms[j + k].setOverride(terms[j + k].getWordsList().indexOf(individualHitTerms[k]), overridePosition);
+                    }
+                }
+            }
         }
+    }
 
-        return joiner.join(largestTerms);
+    private void overrideOutputLongest(WordsAndScores[] terms) {
+        for (WordsAndScores ws : terms) {
+            if (ws == null) {
+                continue;
+            }
+            if (ws.getUseScores()) {
+                return;
+            }
```

Comment:
I would collapse these into a single check, but that's just personal preference.